distribution/lib/Standard/Test/0.0.0-dev/src/Test_Reporter.enso

```diff
@@ -154,6 +176,58 @@ print_group_report group_name test_results config builder =
 
 ## PRIVATE
    Escape Text for XML
-escape_xml : Text -> Text
-escape_xml input =
-    input.replace '&' '&amp;' . replace '"' '&quot;' . replace "'" '&apos;' . replace '<' '&lt;' . replace '>' '&gt;'
+escape_xml : Text -> Boolean -> Text
+escape_xml input inside_attribute=False =
+    escaped = input.replace '&' '&amp;' . replace '"' '&quot;' . replace "'" '&apos;' . replace '<' '&lt;' . replace '>' '&gt;'
+    if inside_attribute then escaped.replace '\n' '&#10;' else escaped
+
+## PRIVATE
+progress_width = 70
+
+## PRIVATE
+print_progress current_progress total_count status_text =
+    total_count_as_text = total_count.to_text
+    counter_width = total_count_as_text.length
+    current_progress_as_text = current_progress.to_text.pad counter_width at=Location.Start
+    line = " ("+ current_progress_as_text + " / " + total_count_as_text + ") " + status_text
+    truncated_line = if line.length <= progress_width then line else
+        line.take (progress_width - 3) + '...'
+
+    Java_System.out.print '\r'
+    Java_System.out.print (' ' * progress_width)
+    Java_System.out.print '\r'
+    Java_System.out.print truncated_line
+    Java_System.out.print '\r'
+
+## PRIVATE
+clear_progress =
+    Java_System.out.print '\r'
+    Java_System.out.print (' ' * progress_width)
+    Java_System.out.print '\r'
+
+## PRIVATE
+type Ignore_Progress_Reporter
```

Comment:
Please ensure that `Ignore_Progress_Reporter` is used on the CI, but the coloring is still there.

Reply:
In 7aeecf2f  I have changed it so that the progress bar is displayed if `System.console() != null`. This should be enough to avoid it running on CI or piped and better than adding yet another env var.

Reply:
I have inspected the runs.

Indeed before the progress was displayed (badly) on CI:
![image](https://github.com/enso-org/enso/assets/1436948/b431f590-23f3-4f1d-9a8f-c7a1a4616c8d)

But after the above commit all seems fine again:
![image](https://github.com/enso-org/enso/assets/1436948/e1ca8e7b-8eb9-477b-9a19-00f8cd8a1f96)


I think the automatic detection of console is even better than an env var.